public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -294,27 +295,71 @@ export class PanelQueryRunner {
    */
   destroy() {
     // Tell anyone listening that we are done
-    if (this.subject) {
-      this.subject.complete();
-    }
-
-    if (this.subscription) {
-      this.subscription.unsubscribe();
-    }
+    this.subject?.complete();
+    this.subscription?.unsubscribe();
+    this.dashboardSubscription?.unsubscribe();
   }
 
   useLastResultFrom(runner: PanelQueryRunner) {
     this.lastResult = runner.getLastResult();
+    this.lastDashboardResult = runner.getLastDashboardResult();
 
     if (this.lastResult) {
       // The subject is a replay subject so anyone subscribing will get this last result
-      this.subject.next(this.lastResult);
+      this.subject.next(this.decorateWithDashboardResults(this.lastResult));
     }
   }
 
   getLastResult(): PanelData | undefined {
     return this.lastResult;
   }
+
+  getLastDashboardResult(): DashboardQueryRunnerResult {
+    return this.lastDashboardResult;
+  }
+
+  refreshDashboardQuerySubscription() {
+    this.initDashboardQuerySubscription();
+  }
+
+  private decorateWithDashboardResults(data: PanelData | undefined): PanelData | undefined {
+    if (!data) {
+      return data;
+    }
+
+    const { alertState, annotations } = this.lastDashboardResult;
+    data.annotations = [new ArrayDataFrame(annotations)];
```

Comment:
won't this overwrite annotations returned from normal query? (the old version did a concat)

Reply:
Good point, what is the scenario when a normal query returns annotations, I've forgotten.